query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java

```diff
@@ -232,6 +234,18 @@ public void purgeIndex(Class<?> entityType) {
       purgeIndex(null, entityType);
    }
 
+   /**
+    * Remove entries from all indexes by key
+    */
+   void removeFromIndexes(TransactionContext transactionContext, Object key) {
+      transactionContext = transactionContext == null ? makeTransactionalEventContext() : transactionContext;
```

Comment:
Indeed, and it looks a bit silly. Let's leave like that in the name of tradition :)